net/isc-dhcp/files/dhcpd.defaults

```diff
@@ -2,10 +2,18 @@
 uci -q get dhcp.isc_dhcpcd && exit 0
 touch /etc/config/dhcp
 
+[ -f /etc/bind/rndc.conf ] && key_secret=$(awk -F'"' '/secret/{print $2}' /etc/bind/rndc.conf | head -1)
+[ -z "$key_secret" ] && key_secret=$(rndc-confgen | awk -F'"' '/secret/{print $2}' | head -1)
+
 uci batch <<EOF
 set dhcp.isc_dhcpd=isc_dhcpd
 set dhcp.isc_dhcpd.authoritative='1'
 set dhcp.isc_dhcpd.default_lease_time='3600'
 set dhcp.isc_dhcpd.max_lease_time='86400'
+set dhcp.dynamicdns=dynamicdns
+set dhcp.dynamicdns.server=127.0.0.1
+set dhcp.dynamicdns.key_algo=hmac-sha256
+set dhcp.dynamicdns.key_secret=$key_secret
+set dhcp.dynamicdns.key_name=local-dns
```

Comment:
What about the scenario where we have multiple DHCP subnets (for different locations or departments or remote workers), but we still have a single centralized DNS server for the entire domain/company?

Reply:
@pprindeville are there any other available and usable algos?

Reply:
See `rndc-confgen(8)`'s `-A` option for documentation.

Reply:
Thank you. I thought they seemed familiar. Same as what bind and knot have available for tsig:
```
Available choices are: hmac-md5, hmac-sha1, hmac-sha224, hmac-sha256, hmac-sha384, and hmac-sha512. The default is hmac-sha256.
```

https://github.com/isc-projects/dhcp/blob/572032cb0e514606559de3784e3f7ca8e1539d17/RELNOTES#L558